File: android/src/main/kotlin/com/kuliahin/size_recommender/SizeRecommender.kt
package com.kuliahin.size_recommender

/**
 * Function for getting size recommendation.
 *
 * @param heightCm - body weight in centimeters.
 * @param weightKg - body weight in kilograms.
 * @return [String] recommended size
 */
fun calculateSizeRecommendation(heightCm: Double, weightKg: Double): String {
    // Validate inputs
    if (heightCm <= 0) throw IllegalArgumentException("Height should be more than 0 cm")
    if (weightKg <= 0) throw IllegalArgumentException("Weight should be more than 0 kg")

    // Convert height from cm to meters
    val heightM = heightCm / 100.0
    val bmi = weightKg / (heightM * heightM)

    // Return result based on calculation
    return when {
        bmi < 18.5 -> "S"
        bmi in 18.5..24.9 -> "M"
        bmi in 25.0..29.9 -> "L"
        else -> "XL"
    }
}

File: ios/Classes/SizeRecommender.swift
import Foundation

public class SizeRecommender {
    
    /// Static function for calculating size recommendation based on BMI.
    ///
    /// - Parameters:
    ///   - heightCm: Height in centimeters.
    ///   - weightKg: Weight in kilograms.
    /// - Returns: Recommended size as a `String`.
    /// - Throws: `NSError` if height or weight is invalid.
    public static func calculateSizeRecommendation(heightCm: Double, weightKg: Double) throws -> String {
        guard heightCm > 0 else { throw NSError(domain: "", code: 1, userInfo: [NSLocalizedDescriptionKey: "Height should be more than 0 cm"]) }
        guard weightKg > 0 else { throw NSError(domain: "", code: 1, userInfo: [NSLocalizedDescriptionKey: "Weight should be more than 0 kg"]) }

        let heightM = heightCm / 100.0
        let bmi = weightKg / (heightM * heightM)

        switch bmi {
        case ..<18.5:
            return "S"
        case 18.5..<25:
            return "M"
        case 25..<30:
            return "L"
        default:
            return "XL"
        }
    }
}
